x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/output/AutodetectResultProcessor.java

```diff
@@ -288,9 +288,15 @@ void processResult(AutodetectResult result) {
             // Commit previous writes here, effectively continuing
             // the flush from the C++ autodetect process right
             // through to the data store
-            bulkResultsPersister.executeRequest();
-            persister.commitResultWrites(jobId);
-            flushListener.acknowledgeFlush(flushAcknowledgement);
+            try {
+                bulkResultsPersister.executeRequest();
+                persister.commitResultWrites(jobId);
```

Comment:
These throwing are admittedly rare, and probably the result of a larger cluster issue, but we should not lock up the flush request. 

It may be better if the flush listener had a `onFlushFailure` logic to capture issues with executing the bulk persistence or index refresh.